### ISSUE TITLE
Printing out non-ASCII X.509 extensions

### ISSUE THREAD

**Issue by richsalz** [openssl/openssl#2378]
Migrated from [rt.openssl.org#3947](https://rt.openssl.org/Ticket/Display.html?id=3947) (status was 'new')

Requestors:
* beldmit@gmail.com

From beldmit@gmail.com on 2015-07-16 21:41:20:
```
Hello,

there is a problem to print out X.509 extensions correctly using the
cmdline utility. There is no way to pass the flags specified by the
"-nameopt" cmdline option to printing callbacks so non-ASCII strings are
always print like "\xD0\x97\xD0\xB0\xD1...".

It concerns, for example, X509_NAME structs in both X509v3 Subject
Alternative Name and X509v3 Authority Key Identifier fields.

Thank you!

-- 
SY, Dmitry Belyavsky
```

**Comment by richsalz** [comment]
is this still an issue, @beldmit ?

**Comment by FdaSilvaYY** [comment]
Fixed in b5c4209be9162d4ceafb9aef833ca94ffa1cc5c9, isn't it ?

**Comment by beldmit** [comment]
I think it is. 
https://github.com/openssl/openssl/commit/b5c4209be9162d4ceafb9aef833ca94ffa1cc5c9 is for nameopt processing, but printing the x509 extensions is not influenced by this flag.

I have a **very** quick-and-dirty works-for-me patch, but it should be seriously adopted to even be demonstrated. It was done for 1.0.2 so it lacks many checks introduced later.

[v3_alt.txt](https://github.com/openssl/openssl/files/1229341/v3_alt.txt)

**Comment by snhenson** [comment]
This is a tricky one. To fix it properly we need a way to pass various options to the extension printing code (e.g the nameopt selection). The API doesn't have a parameter to do that so it would probably require API changes to fix this which rules out OpenSSL 1.1.1.

An alternative is to not allow the full range of name options and instead just use a sensible default which can't be changed by the application. Changing it to print appropriate extensions as "raw" is one way to do that and we could print out otherName that way too instead of the unfriendly "unsupported" output we currently have.

**Comment by beldmit** [comment]
May be it will be enough for the practical purposes just to define a global variable to manage the output mode?

**Comment by nhorman** [comment]
Marking as inactive to be closed at the end of 3.4 dev barring further input

**Comment by beldmit** [comment]
Can be closed, we have a duplicated issue.
